context/src/main/java/io/grpc/Context.java

```diff
@@ -860,6 +868,38 @@ public String toString() {
   }
 
   /**
+   * Builds a context with arbitrary number of values.
+   */
+  public static final class Builder {
+    private final Context parent;
+    private final ArrayList<Object[]> values = new ArrayList<Object[]>();
+
+    private Builder(Context parent) {
+      this.parent = checkNotNull(parent, "parent");
+    }
+
+    /**
+     * Adds a value.
+     *
+     * <p>Like {@code withValues()} on the {@link Context}, the builder doesn't check duplicate
+     * keys. The outcome of duplicate keys given the same builder is unspecified.
+     *
+     * @return this builder
+     */
+    public <K,V> Builder withValue(Key<V> k, V v) {
```

Comment:
`with` is the wrong word here.  Builders use `set` and `add`.  `with` implies that you get a forked copy.